torch/hub.py

```diff
@@ -408,7 +408,7 @@ def get_dir():
     return os.path.join(_get_torch_home(), "hub")
 
 
-def set_dir(d):
+def set_dir(d: Union[str, os.PathLike]) -> None:
```

Comment:
I think this can technically be a bytes string too, (although I presume that would be really rare).

Reply:
I just checked the rest of the repo and `Union[str, os.PathLike]` is the most commonly used way to represent paths. Not opposed to adding `bytes`, just wanted to match the rest of the repo.